components/inputs/input-date-range.js

```diff
@@ -129,16 +149,71 @@ class InputDateRange extends RtlMixin(LocalizeCoreElement(LitElement)) {
 		if (input) input.focus();
 	}
 
+	async validate(showErrors) {
+		const errors = await super.validate(showErrors);
+		if (errors.length !== 0) {
+			return errors;
+		}
+		setTimeout(async() => {
```

Comment:
I added this `setTimeout` because when going from an invalid range to just an invalid input-date, I was getting this error:
```
run-async.js:50 Uncaught (in promise) Error
    at run-async.js:50
    at NodePart.commit (parts.ts:210)
    at TemplateInstance.update (template-instance.ts:53)
    at NodePart.__commitTemplateResult (parts.ts:271)
    at NodePart.commit (parts.ts:221)
    at render (render.ts:55)
    at Function.render (shady-render.ts:277)
    at HTMLElement.update (lit-element.ts:216)
    at HTMLElement.performUpdate (updating-element.ts:757)
    at HTMLElement._enqueueUpdate (updating-element.ts:713)
```

It still functioned properly, but something about the timing seemed right. `setTimeout` was the only thing I had found that "fixed" it.

Reply:
I'm not entirely sure what the root cause is but it seems to be triggered by the icon change in `d2l-input-date`. I believe its an issue with the `runAsync` directive.

You can remove the need for `setTimeout` by changing it from this:
```
<d2l-icon icon="${this.invalid ? 'tier1:alert' : 'tier1:calendar'}" slot="left" style="${styleMap({color: this.invalid ? 'var(--d2l-color-cinnabar)' : ''})}"></d2l-icon>
```
to this:
```
${this.invalid
	? html`<d2l-icon icon="tier1:alert" slot="left" style="${styleMap({ color: 'var(--d2l-color-cinnabar)' })}"></d2l-icon>`
	: html`<d2l-icon icon="tier1:calendar" slot="left"></d2l-icon>`
}
```

Reply:
Thanks for figuring that out!